### ISSUE TITLE
%{recipient.ticket_link} is empty... but template is appended with correct URL in one of the templates

### ISSUE THREAD

**Issue by mrajwadk** [osTicket/osTicket#2180]
hi

We are using  osTicket (v1.9.8)

I am trying to add %{recipient.ticket_link}  to one of the templates for the Agent Email Templates. 
That sting is returned empty.... 

The Client/Recipient email/template also has a link of the format 

Ticket URL : Ticket Link 

where 'Ticket Link' is linked to %{recipient.ticket_link}

however it shows up as...  with no actual hyperlink in the email. 
# 

----- reply above this line ----
Dear XXX,

A request for support has been created and assigned a ticket. A CAD team member will
follow-up with you as soon as possible. Please click on the ticket link to view this
ticket's progress online.

Ticket No    : xppx-001010: 
Ticket Title  : test ticket (hopefully last one)
Ticket URL : [Ticket Link][1]

---

[1] https://www.servername.com/osticket/view.php?auth=o1xa2aaaahsamaaaan6uk9eRdio1yQ%3D%3D
Ref-Mid: $vhoAAA==$be30281880@0cebd7799b
# 

The last line of that email is a functioning link, with auto-logins, and seems to be the 
correct value for %{recipient.ticket_link}, however the variable seems to be empty, 

That link in the client/email is not in the corresponding template, so some other php code is probably
appending that to the email.. 

Any hints ?

**Comment by ntozier** [comment]
If you updating a staff template... wouldn't you want to use %{ticket.staff_link} instead?

**Comment by mrajwadk** [comment]
Yes, that link is already there by default, 
I also want the client/auto-login link... 

We have linked OS ticket with some other legacy system used by some of our other offices, 
and I need to auto-close the tickets there after its passed on to our system. In the process
I need to use %{recipient.ticket_link}   but its empty !! :-(

**Comment by greezybacon** [comment]
The admin email address is intended to be a mailing list for critical errors and system auditing. I think you should just omit sending the new ticket alert to the admin email address and include yourself as a department manager so that the ticket link will work properly

**Comment by mrajwadk** [comment]
Updated the description of the bug to use the word "Agent Email Template" instead of Admin. I do not see the proper definition of %{recipient.ticket_link} in my deployment, both for Staff/Agent templates and also for End-User templates.

**Comment by greezybacon** [comment]
Who is the `recipeint` of the email? Is that the admin email address listed in the system settings page?

**Comment by mrajwadk** [comment]
No... 
My understanding is... 
The recipient is the one who opened the issue/ticket, 
The staff, is the engineer who is supposed to solve the issue/ticket
The admin, is the administrator of the internal site, who has nothing to do with the issue/ticket, opened by the recipient.

**Comment by greezybacon** [comment]
> The recipient is the one who opened the issue/ticket, 

No. That's an EndUser who is the ticket owner in this case. (osTicket also support EndUser collaborators.) The recipient is the individual each to whom each mail is sent. In other words, if creating a ticket triggers an email to an EndUser and four Agents, then there will be five different recipients.

**Comment by zizackov** [comment]
it does not work in 1.10 as well. the user gets %{recipient.ticket_link} in the email link.

**Comment by tsmjhu** [comment]
Hello.  I am also adding that for a newly-installed 1.9.9 system, we are seeing the same as described above: "... the user gets %{recipient.ticket_link} in the email link."  Using the 'Ticket Assignment Alert' template as an example, the other variables (e.g.- %{ticket.number}, %{ticket.name}, %{ticket.subject}, etc.) are filled in fine. e.g. - For the above, "%{ticket.number}" shows the ticket number, but the corresponding hyperlink in the email client simply shows the variable "%{recipient.ticket_link}" itself.  I also have a question, just to clarify... newly-installed, for the Templates that have ticket links, they all initially show as "%{ticket.staff_link}", but, on any of the Templates, when clicking "Supported Variables", "%{ticket.staff_link}" is not listed at all.  So is it correct that "%{recipient.ticket_link}" has in fact replaced the no-longer-used "%{ticket.staff_link}" variable?  In any case, I have tried both (e.g. - manually editing the Template), and neither work, producing the same result.  Just wanted to clarify.  We have a separate/older v1.8.1.2, where the links (%{ticket.staff_link}) are working fine.  This new v1.9.9 is replacing a separate/older v1.9.3, which also worked fine (with %{ticket.staff_link}). Thanks very much.

**Comment by Boduje** [comment]
I think I might have found the answer to this.  Looking at the class.ticket.php file, the "ticket_link" variable seems to have been changed to "client_link".  At least in 1.9.9.

Also I found that it is unnecessary to use the href tag (e.g. <a href="%{ticket.client_link}">Ticket</a>)  You only need the variable (e.g. %{ticket.client_link}

It works for me.  I hope it helps.

**Comment by mrajwadk** [comment]
somebody said this  on the email thread 
"I think I might have found the answer to this. Looking at the class.ticket.php file, the "ticket_link" variable seems to have been changed to "client_link". At least in 1.9.9. Also I found that it is unnecessary to use the href tag (e.g. Ticket) You only need the variable (e.g. %{ticket.client_link} It works for me. I hope it helps."

I do know that works, BUT, it does not have auth-<token> so I am not really looking for that 
With that one needs to login to the site to see that issue/ticket
